module/VuFind/src/VuFind/Content/Covers/Bokinfo.php

```diff
@@ -0,0 +1,173 @@
+ * @package  Content
+ * @author   Jacob Sandin <jacob@imcode.com>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\Content\Covers;
+
+use SimpleXMLElement;
+
+/**
+ * Summon cover content loader.
+ *
+ * @category VuFind
+ * @package  Content
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class Bokinfo extends \VuFind\Content\AbstractCover implements
+    \Laminas\Log\LoggerAwareInterface,
+    \VuFindHttp\HttpServiceAwareInterface
+{
+    use \VuFind\Log\LoggerAwareTrait;
+    use \VuFindHttp\HttpServiceAwareTrait;
+
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        //I will leave this untill I know if they allow it.
+        //$this->cacheAllowed = true;
+        $this->supportsIsbn = true;
+    }
+
+    /**
+     * Get image URL for a particular API key and set of IDs (or false if invalid).
+     *
+     * @param string $key  API key
+     * @param string $size Size of image to load (small/medium/large)
+     * @param array  $ids  Associative array of identifiers (keys may include 'isbn'
+     * pointing to an ISBN object and 'issn' pointing to a string)
+     *
+     * @return string|bool
+     */
+    public function getUrl($key, $size, $ids)
+    {
+        if (!isset($ids['isbn'])) {
+            return false;
+        }
+        if (!isset($key)) {
+            return false;
+        }
+
+        $isbn = $ids['isbn']->get13();
+        try {
+            $client = $this->createHttpClient(
+                "https://api.bokinfo.se/book/get/$isbn"
+            );
+
+            $client->getRequest()->getHeaders()
+                  ->addHeaderLine("Ocp-Apim-Subscription-Key", $key);
+
+            $resp= $client->send();
+            $body = $resp->getBody();
+            $url = $this->getImageUrl($body);
+            if ($this->testUrlFunction($url)) {
+                return "$url";
+            }
+        } catch (\Throwable $ex) {
+            return false;
+        }
+
+        return false;
+    }
+
+    /**
+     * Return a HTTP Client object
+     *
+     * @param string $url API Url
+     *
+     * @return HttpClient Http Client
+     */
+    protected function createHttpClient($url)
+    {
+        $client = $this->httpService->createClient($url);
+
+        $client->setOptions(
+            ['useragent' => 'VuFind', 'keepalive' => true]
+        );
+
+        return $client;
+    }
+
+    /**
+     * Test that the url is realy working
+     *
+     * @param string $url image Url
+     *
+     * @return bool Http Client
+     */
+    protected function testUrlFunction($url)
+    {
+        try {
+            $client = $this->createHttpClient(
+                "$url"
+            );
+
+            $resp= $client->send();
```

Comment:
Were you going to set the method to HEAD here, or did you find that it is not necessary/doesn't work?

Reply:
It actually worked fine without, so therefore I removed it.

Reply:
I'm curious, though, if this is causing VuFind to effectively download every image twice -- once to check if it's valid, and again to cache/display it. Is there a reason that you have to do this? Are you unable to simply trust the results of the API call?